services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts

```diff
@@ -54,25 +54,37 @@ const hasValidContract = (sub: LockedHealthPlanFormDataType): boolean =>
         ))
 
 const hasValidRates = (sub: LockedHealthPlanFormDataType): boolean => {
-    const validBaseRate =
-        sub.rateType !== undefined &&
-        sub.rateDateCertified !== undefined &&
-        sub.rateDateStart !== undefined &&
-        sub.rateDateEnd !== undefined &&
-        (sub.rateProgramIDs !== undefined || sub.rateProgramIDs !== [])
+    const validRates =
+        sub.rateInfos.length === 0
+            ? false
+            : sub.rateInfos.every((rateInfo) => {
+                  const validBaseRate =
+                      rateInfo.rateType !== undefined &&
+                      rateInfo.rateDateCertified !== undefined &&
+                      rateInfo.rateDateStart !== undefined &&
+                      rateInfo.rateDateEnd !== undefined &&
+                      (rateInfo.rateProgramIDs !== undefined ||
+                          rateInfo.rateProgramIDs !== [])
```

Comment:
! == [] will always be true.  I think you want a length check here.

Reply:
You are 100% correct. If we do change this to check for length though, existing submissions in prod will no longer be able to submit without adding rate programs. State users would not know that the submission failed because of missing rate program sunless they navigate back to `Rate Details` page. 

We could fix with multiple ways like our migrations tool, front end logic or backend logic, but we should separate this into another ticket, [MR-2680](https://qmacbis.atlassian.net/browse/MR-2680),  and combine the fix with addressing existing submissions